src/schema/common.cljs

```diff
@@ -1,7 +1,89 @@
 (ns schema.common
   (:require
+    [malli.core :as malli]
+    malli.util
     [schema.registry :as registry]))
 
+(defn- optional-keys
```

Comment:
FYI: The function `optional-keys` exists also exists in malli's source https://github.com/metosin/malli/blob/9f5b8095e1bc72b04ce8c006ae8951b73a0afcf9/src/malli/util.cljc#L240. The problem is that, for the developer to ignore certain keys, they need to pass a set of keys. Our solution is different, each child key of `:schema.common/map` can have a property `:no-optional` that overrides the parent property `:optional` and I believe this is much more convenient for our use cases.